front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/NotificationControllerTck.groovy

```diff
@@ -188,15 +189,17 @@ class S3NotificationControllerTck extends NotificationControllerTck {
   def scheduler = Schedulers.from(Executors.newFixedThreadPool(1))
 
   @Shared
-  S3NotificationDAO s3NotificationDAO
+  NotificationDAO notificationDAO
 
   @Override
   NotificationDAO createNotificationDAO() {
     def amazonS3 = new AmazonS3Client(new ClientConfiguration())
     amazonS3.setEndpoint("http://127.0.0.1:9999")
     S3TestHelper.setupBucket(amazonS3, "front50")
 
-    s3NotificationDAO = new S3NotificationDAO(new ObjectMapper(), amazonS3, scheduler, 0, "front50", "test")
-    return s3NotificationDAO
+    def storageService = new S3StorageService(new ObjectMapper(), amazonS3, "front50", "test")
```

Comment:
I think the right thing to do here is to write the tests in terms of the StorageService notification so that you are testing the model is correct for the data model / resource implementations themselves.

Then we need to write a unit test suite to test the specialized StorageService implementations. As I mentioned before, ideally there is a base suite that we inject a concrete class into and it verifies the contract. Each service can add additional tests beyond that for the extra/special behaviors on it. That's what I've done in the past for similar situations and it turns out pretty easy. Except here there is the extra underlying service calls that would need to be mocked with spock so there's a twist that might not be so straightforward to reuse a standard test suite.